src/estimagic/inference/bootstrap.py

```diff
@@ -51,88 +57,70 @@ def bootstrap(
         BootstrapResult: A BootstrapResult object storing information on summary
             statistics, the covariance matrix, and estimated boostrap outcomes.
     """
-    check_inputs(data, cluster_by)
-
-    if outcome_kwargs is not None:
-        outcome = functools.partial(outcome, **outcome_kwargs)
-
-    bootstrap_outcomes = get_bootstrap_outcomes(
-        data=data,
-        outcome=outcome,
-        cluster_by=cluster_by,
-        seed=seed,
-        n_draws=n_draws,
-        n_cores=n_cores,
-        error_handling=error_handling,
-        batch_evaluator=batch_evaluator,
-    )
-
-    base_outcome = outcome(data)
-    out = bootstrap_from_outcomes(base_outcome, bootstrap_outcomes)
-
-    return out
-
-
-def bootstrap_from_outcomes(base_outcome, bootstrap_outcomes):
-    """Create BootstrapResults object.
-
-    Args:
-        base_outcome (pytree): Pytree of base outcomes, i.e. the outcome
-            statistic(s) evaluated on the original data set.
-        bootstrap_outcomes (list): List of pytrees of estimated
-            bootstrap outcomes.
+    if existing_outcomes is None:
+        existing_outcomes = []
+    elif not isinstance(existing_outcomes, list):
+        raise ValueError("existing_outcomes must be a list.")
+
+    if callable(outcome):
+
+        check_inputs(data=data, cluster_by=cluster_by)
+
+        if outcome_kwargs is not None:
+            outcome = functools.partial(outcome, **outcome_kwargs)
+
+        bootstrap_outcomes = get_bootstrap_outcomes(
+            data=data,
+            outcome=outcome,
+            cluster_by=cluster_by,
+            seed=seed,
+            n_draws=n_draws,
+            n_cores=n_cores,
+            error_handling=error_handling,
+            batch_evaluator=batch_evaluator,
+        )
+        existing_outcomes += bootstrap_outcomes
```

Comment:
I think this would change `existing_outcomes` in place. Also, I suggest to give the combined outcomes a name that reflects that they are already combined. i.e. something like `all_outcomes = existing_outcomes + bootstrap_outcomes`

Reply:
Very good catch! Much better now.